plugin/evm/limitorders/contract_events_processor.go

```diff
@@ -110,11 +109,17 @@ func (cep *ContractEventsProcessor) handleOrderBookEvent(event *types.Log, remov
 		log.Info("HandleOrderBookEvent", "OrderCancelled args", args, "removed", removed)
 		orderId := parseOrderId(args["orderHash"])
 		if !removed {
-			cep.database.GetOrderBookData().OrderMap[orderId].Status = Cancelled
+			if cep.database.setOrderStatus(orderId, Cancelled) != nil {
```

Comment:
`if err = cep.database.setOrderStatus(orderId, Cancelled); err != nil {`

is a better way to write(go idiomatic way to write)